Wikipedia/Code/UIImageView+WMFImageFetchingInternal.m

```diff
@@ -187,6 +190,14 @@ - (void)wmf_setImage:(UIImage *)image
     };
     
     self.image = image;
+    
+    if (image.isGIF && [self isKindOfClass:[FLAnimatedImageView class]]) {
+        NSURL *imageURL = [self wmf_imageURLToFetch];
+        FLAnimatedImage *animatedImage = [FLAnimatedImage animatedImageWithGIFData: [[self wmf_imageController] diskDataForImageWithURL:imageURL]];
```

Comment:
Since `diskDataForImageWithURL:` could return nil (even though it should exist in pretty much every case) this should be nil checked. Also, I'm not sure if `animatedImageWithGIFData` will return nil if it fails, so it would probably good to check that for nil as well. If either fails, leave the original image.